app/failed_logins_by_ip/rest.py

```diff
@@ -0,0 +1,83 @@
+from flask import Blueprint, jsonify
+
+from app.dao.failed_logins_by_ip_dao import (
+    dao_create_failed_login_for_ip,
+    dao_get_count_of_all_failed_logins_for_ip,
+    dao_get_failed_logins,
+    dao_get_latest_failed_login_by_ip,
+)
+from app.errors import InvalidRequest, register_errors
+from app.utils import (
+    calculate_delay_period,
+    check_request_within_throttle_period,
+    get_ip_address,
+)
+
+failed_logins_by_ip_blueprint = Blueprint(
+    "failed_logins",
+    __name__,
+    url_prefix="/failed-logins",
+)
+
+register_errors(failed_logins_by_ip_blueprint)
+
+
+@failed_logins_by_ip_blueprint.route("fetch-all")
+def get_all_failed_logins():
+    """
+    Fetches all failed login records in the table.
+    """
+    data = dao_get_failed_logins()
+    data = [obj.serialize() for obj in data] if data else []
+    return jsonify(data or {}), 200
+
+
+@failed_logins_by_ip_blueprint.route("fetch-for-requester-ip")
+def get_failed_login_by_ip():
```

Comment:
possibly change signature to:
@failed_logins_by_ip_blueprint.route("fetch-for-requester")
def get_failed_login_for_requester()...

The current route/method names suggest that an IP can be supplied to fetch the failed login for any IP - may cause confusion for future devs using/modifying the code.